core/src/ldml/ldml_transforms.cpp

```diff
@@ -7,15 +7,262 @@
 
 #include "ldml_transforms.hpp"
 #include "debuglog.h"
+#include <algorithm>
+#include <string>
 
 #ifndef assert
-#define assert(x) // TODO-LDML
+#define assert(x)  // TODO-LDML
 #endif
 
 namespace km {
 namespace kbp {
 namespace ldml {
 
+element::element(const USet &new_u, KMX_DWORD new_flags)
+    : chr(), uset(new_u), flags((new_flags & ~LDML_ELEM_FLAGS_TYPE) | LDML_ELEM_FLAGS_TYPE_USET) {
+}
+
+element::element(km_kbp_usv ch, KMX_DWORD new_flags)
+    : chr(ch), uset(), flags((new_flags & ~LDML_ELEM_FLAGS_TYPE) | LDML_ELEM_FLAGS_TYPE_CHAR) {
+}
+
+bool
+element::is_uset() const {
+  return (flags & LDML_ELEM_FLAGS_TYPE) == LDML_ELEM_FLAGS_TYPE_USET;
+}
+
+signed char
+element::get_order() const {
+  unsigned char uorder = ((flags & LDML_ELEM_FLAGS_ORDER_MASK) >> LDML_ELEM_FLAGS_ORDER_BITSHIFT);
+  return (signed char)uorder;
+}
+
+signed char
+element::get_tertiary() const {
+  unsigned char uorder = ((flags & LDML_ELEM_FLAGS_TERTIARY_MASK) >> LDML_ELEM_FLAGS_TERTIARY_BITSHIFT);
+  return (signed char)uorder;
+}
+
+bool
+element::is_prebase() const {
+  return flags & LDML_ELEM_FLAGS_PREBASE;
+}
+
+bool
+element::is_tertiary_base() const {
+  return flags & LDML_ELEM_FLAGS_TERTIARY_BASE;
+}
```

Comment:
It'd be good for these to return true bools rather than truthy values. Just avoids possible frustrations in future.

```suggestion
bool
element::is_prebase() const {
  return !!(flags & LDML_ELEM_FLAGS_PREBASE);
}

bool
element::is_tertiary_base() const {
  return !!(flags & LDML_ELEM_FLAGS_TERTIARY_BASE);
}
```